tests/suites/test_suite_ssl.function

```diff
@@ -3680,6 +3737,44 @@ void handshake_serialization( )
 }
 /* END_CASE */
 
+/* BEGIN_CASE depends_on:MBEDTLS_X509_CRT_PARSE_C:!MBEDTLS_USE_PSA_CRYPTO:MBEDTLS_PKCS1_V15:MBEDTLS_RSA_C:MBEDTLS_ECP_DP_SECP384R1_ENABLED */
+void handshake_fragmentation( int mfl, int expected_srv_hs_fragmentation, int expected_cli_hs_fragmentation)
+{
+    handshake_test_options options;
+    log_pattern srv_pattern, cli_pattern;
+    size_t srv_hs_fragmented = 0;
+    size_t cli_hs_fragmented = 0;
+
+    srv_pattern.pattern = cli_pattern.pattern = "found fragmented DTLS handshake";
+    srv_pattern.counter = &srv_hs_fragmented;
+    cli_pattern.counter = &cli_hs_fragmented;
+
+    init_handshake_options( &options );
+    options.dtls = 1;
+    options.mfl = mfl;
+    options.srv_auth_mode = MBEDTLS_SSL_VERIFY_REQUIRED;
+    options.srv_log_obj = &srv_pattern;
+    options.cli_log_obj = &cli_pattern;
+    options.srv_log_fun = log_analyzer;
+    options.cli_log_fun = log_analyzer;
+
+    perform_handshake( &options );
+
+    /* Test if the server received fragmented handshake */
+    if( expected_srv_hs_fragmentation )
+    {
+        TEST_ASSERT( srv_hs_fragmented >= 1 );
+    }
+    /* Test if the client received fragmented handshake */
+    if( expected_cli_hs_fragmentation )
+    {
+        TEST_ASSERT( cli_hs_fragmented >= 1 );
+    }
+    /* The goto below is used to avoid an "unused label" warning.*/
+    goto exit;
```

Comment:
This `goto` seems weird. Why do we need it here but not in other tests?

Reply:
Yes, this `goto exit;` is unnecessary here because it is already in TEST_ASSERT. In the other tests it is needed to prevent warning. After Andrzej's refactoring some of test does not have any TEST_ASSERT, but this `exit:` label is added automatically during compilation (by script). I will remove it

Reply:
removed